### PR TITLE
Updated templates using Panel template + React-grid-layout

### PR DESCRIPTION
**Pull request by AjayThorve**
This PR introduces:

1. react-grid-layouts for dashboards via the panel template engine. 
2. `layout_array` input param for dashboard, which is a nested list for a customized dashboard layout. 

```python
d = cux_df.dashboard(charts_list[:4], theme=cuxfilter.themes.dark, title="Layout - single feature",
                     layout_array=
                     [
                         [1,1,2],
                         [3,3,2]
                     ]
)
```
would result in something like:
image

3. `Loading animation` when python callbacks are being executed, thanks to panel templates.
4. CSS based themes - currently dark and light. A follow-up PR would refactor the python `apply_themes` methods.
5. Wrapped all charts in a `Card` view, that allows minimizing, dragging and resizing the card

### CODE REVIEW

File: python/cuxfilter/layouts/layouts.py [hunk 1]
```diff
@@ -25,234 +13,185 @@ def is_widget(obj):
...
+                            np.where(arr == i + 1)[0][-1] + 1,
+                            np.where(arr == i + 1)[1][-1] + 1,
+                        ]
+                    )
+                    / np.array(arr.shape)
+                    * (6, 12)
+                )
+                tmpl.main[int(x0) : int(x1), int(y0) : int(y1)] = plots[
+                    i
+                ].view()
```
Comment (bryevdv) [inline python/cuxfilter/layouts/layouts.py:74]: suggest defining a helper to condense these codepaths, e.g similar to
```
def _helper(array, i, pos):
    x, y = np.array([
        np.where(arr == i + 1)[0][pos],
        np.where(arr == i + 1)[1][pos],
    ]) / np.array(arr.shape) * (6, 12)
    return int(x), int(y)
```

Then the above can be streamlined
```
    if i < len(plots):
        x0, y0 = _helper(arr, i, 0)
        x1, y1 = _helper(arr, i, -1) 
        tmpl.main[x0:x1, y0:y1] = plots[i].view()
```

**Review by bryevdv — CHANGES_REQUESTED** [review]

**Comment by bryevdv** [comment]
only a few small comments, this is a great simplification!

File: python/cuxfilter/layouts/layouts.py (hunk 1 above)
Comment (AjayThorve) [reply python/cuxfilter/layouts/layouts.py]: thanks, implemented this

File: python/cuxfilter/layouts/custom_react_template.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    main = param.ClassSelector(
+        class_=GridSpec,
+        constant=True,
+        doc="""
+        A list-like container which populates the main area.""",
+    )
+
+    row_height = param.Integer(default=0)
+
+    _css = pathlib.Path(__file__).parent / "assets/react.css"
```
[the remark this answers is not in the document]
Comment (AjayThorve) [reply python/cuxfilter/layouts/custom_react_template.py]: fixed

File: python/cuxfilter/charts/datashader/custom_extensions/interactive_image.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 
 def bokeh_notebook_div(image):
-    """"
+    """ "
```
[the remark this answers is not in the document]
Comment (AjayThorve) [reply python/cuxfilter/charts/datashader/custom_extensions/interactive_image.py]: removed

File: python/cuxfilter/charts/bokeh/plots.py [hunk 4]
```diff
@@ -282,7 +280,7 @@ def generate_chart(self):
         generate chart
         """
         self.chart = figure(
-            title=self.title,
+            # title=self.title,
```
[the remark this answers is not in the document]
Comment (AjayThorve) [reply python/cuxfilter/charts/bokeh/plots.py]: removed

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
nice. The `panel` dependency will need to be updated in the `integration` repo below as well.

https://github.com/rapidsai/integration/blob/branch-0.18/conda/recipes/versions.yaml#L111-L112

**Comment by AjayThorve** [comment]
> nice. The `panel` dependency will need to be updated in the `integration` repo below as well.
> 
> https://github.com/rapidsai/integration/blob/branch-0.18/conda/recipes/versions.yaml#L111-L112

Thanks! I created a PR - https://github.com/rapidsai/integration/pull/209

**Comment by AjayThorve** [comment]
rerun tests

**Comment by AjayThorve** [comment]
CI failing - waiting for  rapidsai/integration#209 to be merged for panel version to be updated

**Review by ajschmidt8 — APPROVED** [review]
approved since https://github.com/rapidsai/integration/pull/209 is now merged.

**Comment by AjayThorve** [comment]
Rerun tests

**Review by bryevdv — APPROVED** [review]

**Comment by AjayThorve** [comment]
@gpucibot merge

**Pull request merged**